internal/resolver/dns/dns_resolver.go

```diff
@@ -326,8 +329,10 @@ func (d *dnsResolver) lookup() (*resolver.State, error) {
 	if hostErr != nil && (srvErr != nil || len(srv) == 0) {
 		return nil, hostErr
 	}
-	state := &resolver.State{
-		Addresses: append(addrs, srv...),
+
+	state := &resolver.State{Addresses: addrs}
+	if len(srv) > 0 {
+		*state = statedata.Set(*state, &statedata.StateData{BalancerAddresses: srv})
```

Comment:
How many copies of `state` does this line make?
Is it better to change the line above to not take the pointer?

Reply:
Done